src/components/TopBar/AboutDialog.js

```diff
@@ -0,0 +1,356 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v.2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import {
+    Alert,
+    AlertTitle,
+    Avatar,
+    Box,
+    Button,
+    Card,
+    CardContent,
+    CardHeader,
+    Chip,
+    CircularProgress,
+    Collapse,
+    Dialog,
+    DialogActions,
+    DialogContent,
+    DialogTitle,
+    Fade,
+    Grid,
+    Skeleton,
+    Tooltip,
+    useMediaQuery,
+    useTheme,
+} from '@mui/material';
+import {
+    Bookmark,
+    BrokenImage,
+    DnsOutlined,
+    Gavel,
+    QuestionMark,
+    WidgetsOutlined,
+} from '@mui/icons-material';
+import { FormattedMessage, useIntl } from 'react-intl';
+import PropTypes from 'prop-types';
+
+const ComponentsTypeIcons = {
+    app: <WidgetsOutlined />,
+    server: <DnsOutlined />,
+    other: <QuestionMark />,
+};
+
+const AboutDialog = ({
+    open,
+    onClose,
+    getGlobalVersion,
+    getLogoThemed,
+    appVersion,
+    appLicense,
+    getAdditionalComponents,
+}) => {
+    const theme = useTheme();
+    const intl = useIntl();
+
+    //TODO is useCallback in component or in caller?
+    const logo = useMemo(() => {
+        if (getLogoThemed) {
+            return getLogoThemed(theme.palette.mode);
+        } else {
+            return <BrokenImage />;
+        }
+    }, [getLogoThemed, theme.palette.mode]);
+
+    const handlerGetGlobalVersion = useCallback(getGlobalVersion, [
+        getGlobalVersion,
+    ]);
+    const [loadingGlobalVersion, setLoadingGlobalVersion] = useState(false);
+
+    /* We want to get the initial version once at first render to detect later a new deploy */
+    const [startingGlobalVersion, setStartingGlobalVersion] = useState(null);
+    useEffect(() => {
+        if (getGlobalVersion) {
+            handlerGetGlobalVersion((value) => {
+                setLoadingGlobalVersion(false);
+                setStartingGlobalVersion(value);
+                setActualGlobalVersion(value);
+            });
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
we never disable this rule, why do you need to ?

Reply:
because it don't like that I don't put in `deps` array all handlers that I use, but it's wanted in my case because I want to run one _only_ at first creation and others two each time the dialog is open only.

Reply:
In which case exactly it cause a problem ?
You just have to add `getGlobalVersion` and its handler, it never change no ?

Reply:
No I want to run it once when the application is loaded/started only. It's to detect later when the dialog is shown if the deployed version has change or not to enable an alert.

Reply:
I don't understand, I logged to be sure, and adding the deps don't change anything.
What case exactly cause a problem ?